### ISSUE TITLE
After upgrading flutter to 1.12 and the cached_network_image to ^2.0.0-rc.1 flutter packages get returns version solving failed. 

### ISSUE THREAD

**Issue by thoffart** [johnpryan/flutter_map#473]
After upgrading flutter to 1.12 and the cached_network_image to ^2.0.0-rc.1, flutter packages get returns version solving failed. 
```
Running "flutter pub get" in gestor-mobile...                   
Because flutter_map 0.7.3 depends on cached_network_image ^1.1.0 and no versions of flutter_map match >0.7.3 <0.8.0, flutter_map ^0.7.3 requires cached_network_image ^1.1.0.

So, because gestor depends on both cached_network_image ^2.0.0-rc.1 and flutter_map ^0.7.3, version solving failed.
pub get failed (1; So, because gestor depends on both cached_network_image ^2.0.0-rc.1 and flutter_map ^0.7.3, version solving failed.)
exit code 1
```

**Comment by lacu** [comment]
me too, after update to 1.12.13 +hotfix.5
`
  ImageStreamCompleter load(NetworkImageWithRetry key) {
                       ^
/e:/flutter/packages/flutter/lib/src/painting/image_provider.dart:403:24: Context: This is the overridden method ('load').
  ImageStreamCompleter load(T key, DecoderCallback decode);
                       ^
/e:/flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-1.1.3/lib/src/cached_network_image_provider.dart:42:24: Error: The method 'CachedNetworkImageProvider.load' has fewer positional arguments than those of overridden method 'ImageProvider.load'.
  ImageStreamCompleter load(CachedNetworkImageProvider key) {
                       ^
/e:/flutter/packages/flutter/lib/src/painting/image_provider.dart:403:24: Context: This is the overridden method ('load').
  ImageStreamCompleter load(T key, DecoderCallback decode);
                       ^
/e:/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_map-0.7.3/lib/src/layer/tile_provider/mbtiles_image_provider.dart:90:24: Error: The method 'MBTileImage.load' has fewer positional arguments than those of overridden method 'ImageProvider.load'.
  ImageStreamCompleter load(MBTileImage key) {
                       ^
/e:/flutter/packages/flutter/lib/src/painting/image_provider.dart:403:24: Context: This is the overridden method ('load').
  ImageStreamCompleter load(T key, DecoderCallback decode);
                       ^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
build failed.

FAILURE: Build failed with an exception.``

**Comment by yobniares** [comment]
Same issue. I had this before and handled with it somehow, don't remember.

**Comment by amirmi17** [comment]
I have same issue and for now this changes in 'pubspec.yaml' file work for me:
```

flutter_map: 
   git:
     url: git://github.com/okaxaki/flutter_map.git
     ref: fix/support-flutter-1.10
```
and
```
flutter_map_marker_cluster:
   git:
      url: git://github.com/emidiao/flutter_map_marker_cluster.git
      ref: e1b4ad9c6d300649acd448a9ba9069d865845686
```

**Comment by johnpryan** [comment]
This should be fixed in 0.8.0. See #480 for details.
